lib/queue.rb

```diff
@@ -1,31 +1,42 @@
 class Queue
 
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @store = Array.new(20)
+    @head = 0
+    @tail = 0
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if @tail == @head && !@store[@head].nil?
+    raise "Buffer is full"
+    end
+    @store[@tail] = element
+    @tail = (@tail + 1) %20
```

Comment:
Suggestion for less hardcoded lengths. 

```suggestion
    @tail = (@tail + 1) % @store.length
```